docs/src/main/paradox/scala/http/server-side/websocket-support.md

```diff
@@ -116,6 +115,20 @@ In case you need to keep inactive connections alive, you can either tweak your i
 'keep-alive' messages regularly.
 @@@
 
+### Handling websockets with independent Source and Sink
+
+Sometimes it is interesting to be able to expose an independent Source and Sink to such WebSocket handler rather than a linear Flow. This is doable by using the `Flow.fromSinkAndSource` or `Flow.fromSinkAndSourceCoupled` methods in Akka 2.5.x (or `CoupledTerminationFlow` in Akka 2.4.x), and passing such composite Flow into the @java[`handleWebSocketRequestWith`]@scala[`handleWebSocketRequestWith`] method. Using these methods you can for example use a `Sink.foreach` to side-efect or print incoming user messages, while providing some kind of data source via a `Source` to the outgoing side of the connection. 
+
+The difference between the "coupled" Flow from Sink and Source and the "normal one", is that the termination of one side of the flow (e.g. the Source completing) will cause the other side to be terminated as well. This is often useful when we want to terminate the connection once the data-source has finished streaming or the user has done away. If unsure about the semantics you require, the "coupled" flow is usually a safe bet, since it will close the connection regardless of which side of the flow terminates, which very often is the desired semantics.
+
+Convenience versions of the handle methods are also available. Specifically, when ignoring incoming mesages from the client-side it is important to be aware that those messages could be `Streamed`, meaning that a plain `Sink.ignore` is not always safe for ignoring the incoming data. For that reason the convenience methods use the `WebSocket.ignoreSink` that actively drains incoming streamed messages. One may also use it explicitly:
```

Comment:
I'm not sure this paragraph makes it clear why/when `Sink.ignore` is unsafe: what does it mean that "those messages could be `Streamed`" and why does that cause problems? Might be good to elaborate on that a bit more here - perhaps first show the code examples and then explain the details about why it is really needed to do it like that?

Reply:
I'll link to the implications of streaming from here perhaps then